src/commands/new.ts

```diff
@@ -0,0 +1,182 @@
+import isIgniteDirectory from '../lib/is-ignite-directory'
+import exitCodes from '../lib/exit-codes'
+import * as path from 'path'
+import addEmptyBoilerplate from '../lib/add-empty-boilerplate'
+import { isEmpty, match, not, toLower } from 'ramda'
+import { IgniteToolbox } from '../types'
+import boilerplateInstall from '../lib/boilerplate-install'
+
+/**
+ * Creates a new ignite project based on an optional boilerplate.
+ */
+module.exports = {
+  alias: ['n'],
+  description: 'Generate a new React Native project with Ignite CLI.',
+  run: async function command(toolbox: IgniteToolbox) {
+    const { parameters, strings, print, filesystem, ignite, prompt, runtime } = toolbox
+    const { isBlank, upperFirst, camelCase } = strings
+    const { log } = ignite
+
+    // grab the project name
+    const projectName = (parameters.first || '').toString()
+
+    // check for kebabs
+    const isKebabCase = not(isEmpty(match(/.-/g, `${projectName}`)))
+
+    // camelCase the project name for user example
+    const projectNameCamel = upperFirst(camelCase(projectName))
+
+    // check for numbers-only names
+    const isNumericOnly = /^\d+$/.test(projectName)
+
+    // check for alphanumeric name, beginning with a letter
+    const isValidName = /^[a-z_][a-z0-9_]+$/i.test(projectName)
+
+    // ensure we're in a supported directory
+    if (isIgniteDirectory(process.cwd())) {
+      print.error('The `ignite new` command cannot be run within an already ignited project.')
+      process.exit(exitCodes.NOT_IGNITE_PROJECT)
+    }
+
+    // prevent installing when node_modules/react-native exists
+    if (filesystem.exists('node_modules/react-native')) {
+      print.error(
+        'The `ignite new` command cannot be run within a directory with `node_modules/react-native` installed.',
+      )
+      print.error('Try installing from a directory without a `node_modules` directory.')
+      process.exit(exitCodes.EXISTING_REACT_NATIVE)
+    }
+
+    // verify the project name is a thing
+    if (isBlank(projectName)) {
```

Comment:
Not a big deal, but why not do each check and exit after its corresponding test (e.g. this would go after line 21). Think of the milliseconds users would save!